eessi/testsuite/tests/apps/lammps/lammps.py

```diff
@@ -0,0 +1,163 @@
+"""
+This module tests the binary 'lmp' in available modules containing substring 'LAMMPS'.
+The tests come from the lammps github repository (https://github.com/lammps/lammps/)
+"""
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+from eessi.testsuite import hooks, utils
+from eessi.testsuite.constants import *  # noqa
+
+
+class EESSI_LAMMPS_base(rfm.RunOnlyRegressionTest):
+    scale = parameter(SCALES.keys())
+    valid_prog_environs = ['default']
+    valid_systems = ['*']
+    time_limit = '30m'
+    device_type = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
+
+    # Parameterize over all modules that start with LAMMPS
+    module_name = parameter(utils.find_modules('LAMMPS'))
+
+    # Set sanity step
+    @deferrable
+    def assert_lammps_openmp_treads(self):
+        '''Assert that OpenMP thread(s) per MPI task is set'''
+        n_threads = sn.extractsingle(
+            r'^  using (?P<threads>[0-9]+) OpenMP thread\(s\) per MPI task', self.stdout, 'threads', int)
+        utils.log(f'OpenMP thread(s) is {n_threads}')
+
+        return sn.assert_eq(n_threads, self.num_cpus_per_task)
+
+    @deferrable
+    def assert_lammps_processor_grid(self):
+        '''Assert that the processor grid is set correctly'''
+        grid = list(sn.extractall(
+            '^  (?P<x>[0-9]+) by (?P<y>[0-9]+) by (?P<z>[0-9]+) MPI processor grid', self.stdout, tag=['x', 'y', 'z']))
+        n_cpus = int(grid[0][0]) * int(grid[0][1]) * int(grid[0][2])
+
+        return sn.assert_eq(n_cpus, self.num_tasks)
+
+    @deferrable
+    def assert_run(self):
+        '''Assert that the test calulated the right number of neighbours'''
+        regex = r'^Loop time of (?P<perf>[.0-9]+) on [0-9]+ procs for 100 steps with (?P<atoms>\S+) atoms'
+        n_atoms = sn.extractsingle(regex, self.stdout, 'atoms', int)
+
+        return sn.assert_eq(n_atoms, 32000)
+
+    @performance_function('img/s')
+    def perf(self):
+        regex = r'^(?P<perf>[.0-9]+)% CPU use with [0-9]+ MPI tasks x [0-9]+ OpenMP threads'
```

Comment:
this doesn't look right, this is the %CPU usage.

performance should be one of the following:
Performance: 0.823 ns/day, 29.175 hours/ns, 4.761 timesteps/s, 152.338 katom-step/s

Reply:
Performance: 205379.307 tau/day, 475.415 timesteps/s, 15.213 Matom-step/s this than?

Reply:
your line comes from the `lj` test, while mine comes from the `rhodo` test.
let's take  `timesteps/s`, as this unit is available for both tests?

Reply:
I've now added `tau/day` for `lj` and ´ns/day` for `rhodo` but I can also take the timesteps

Reply:
i just checked and both `tau/day` for lj and `ns/day` for rhodo scale in exactly the same way as `timesteps/s`, so it doesn't really matter.

i do have a slight preference for `timesteps/s` as it is easy to understand. otherwise, can add a comment what exactly `tau/day` means?

Reply:
I also do not know what `tau/day` is and google does not have a ready explanation. So changed it to the `timesteps/s` for `lj` and `rhodo`